### ISSUE TITLE
Can you provide the training script to reproduce the experiment?

### ISSUE THREAD

**Issue by cjz007** [inclusionAI/ASearcher#10]
Hi.

The training script seems incomplete and many hyperparams seems to be inconsitent to the paper. Like the max_new_token is 1024 in the config file, but in the paper it is much longer

**Comment by samjia2000** [comment]
> The training script seems incomplete and many hyperparams seems to be inconsitent to the paper. Like the max_new_token is 1024 in the config file, but in the paper it is much longer

@cjz007 Hi, the `max_new_tokens` is the maximum generation length for each turn. In total the model is allowed to output `max_turns` x `max_new_tokens` tokens in one trajectory.

**Comment by samjia2000** [comment]
About the training steps: the Qwen2.5 7B result is taken at step 240. It is not required to train for 10 epochs.

About GRPO:  As the training side does not recognize "trajectories". The group normalization operation is done in a hacking manner. Specifically, the minus of average group rewards is done in [AReaL/ASearcher/train/asearcher.py#L191](https://github.com/inclusionAI/ASearcher/blob/20b27b2efeb062828752dcee8273389ee4d46de2/AReaL/ASearcher/train/asearcher.py#L191).  Setting `adv_norm` as True normalizes the values. This is actually a little bit different from the standard GRPO implementation but the training 

About the dumped trajectories: no.  If you do not want to dump trained trajectories, you can remove the code for dumping trajectories.

ppo_n_minibatches: in practice we find `ppo_n_minibatches=1` converges to better results. You can try setting `ppo_n_minibatches>1` for ablation.
